File: app/src/main/java/com/isl/leaseManagement/activities/Notification/NotificationDetailActivity.java
package com.isl.leaseManagement.activities.Notification;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.isl.leaseManagement.utils.CustomTextView;

import infozech.itower.R;

public class NotificationDetailActivity  extends AppCompatActivity {
    CustomTextView txview_notification,txtview_siteid,txtview_requestid,txtview_task;
    ImageView img_back;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_notification_detail);
        txview_notification = findViewById(R.id.txview_notification);
        txtview_siteid = findViewById(R.id.txtview_siteid);
        txtview_requestid = findViewById(R.id.txtview_requestid);
        txtview_task = findViewById(R.id.txtview_task);
        img_back = findViewById(R.id.img_back);
        img_back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                finish();
            }
        });
        getIntentValues();
    }
    public void getIntentValues()
    {
        Intent intent = getIntent();
        if(intent != null)
        {
            String notificationSubject = intent.getStringExtra("subject");
            String[] siteId = intent.getStringExtra("siteid").split(":");
            String[] requestId = intent.getStringExtra("requestid").split(":");
            String[] task = intent.getStringExtra("task").split(":");
            txview_notification.setText(notificationSubject);
            txtview_siteid.setText(siteId[1]);
            txtview_requestid.setText(requestId[1]);
            txtview_task.setText(task[1]);
        }
    }
}

File: app/src/main/java/com/isl/leaseManagement/paymentProcess/activities/notification/NotificationDetailActivity.kt
package com.isl.leaseManagement.paymentProcess.activities.notification

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.isl.itower.MyApp
import com.isl.leaseManagement.room.db.MyDatabase
import com.isl.leaseManagement.utils.CustomTextView
import infozech.itower.R
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class NotificationDetailActivity : AppCompatActivity() {

    var txview_notification: CustomTextView? = null
    var txtview_siteid: CustomTextView? = null
    var txtview_requestid: CustomTextView? = null
    var txtview_task: CustomTextView? = null
    var db: MyDatabase? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        db = MyApp.getMyDatabase();
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_notification_detail)
        txview_notification = findViewById(R.id.txview_notification)
        txtview_siteid = findViewById(R.id.txtview_siteid)
        txtview_requestid = findViewById(R.id.txtview_requestid)
        txtview_task = findViewById(R.id.txtview_task)
        var img_back: ImageView? = findViewById(R.id.img_back)
        img_back?.setOnClickListener(View.OnClickListener { finish() })
        intentValues
    }

    val intentValues: Unit
        get() {
            val intent = intent
            var id: Int = 0;

            if (intent != null) {
                val notificationSubject = intent.getStringExtra("subject")
                val siteId = intent.getStringExtra("siteid")!!.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                val requestId = intent.getStringExtra("requestid")!!.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                val task = intent.getStringExtra("task")!!.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                //txview_notification.setText(notificationSubject);
                id = intent.getStringExtra("id")!!.toInt()
                txtview_siteid!!.text = siteId[1]
                txtview_requestid!!.text = requestId[1]
                txtview_task!!.text = task[1]
            }

            if (id!=0 && id!=-1){
                lifecycleScope.launch(Dispatchers.IO) {
                    db?.notificationDao()?.markAsReadById(id);
                }
            }
        }
}
